src/MobileBookingHeader/OneWayTrip.js

```diff
@@ -0,0 +1,47 @@
+// @flow
+
+import * as React from 'react';
+import { createFragmentContainer, graphql } from 'react-relay';
+import idx from 'idx';
+
+import type OneWayTrip_booking from './__generated__/OneWayTrip_booking.graphql';
+import { TripDescriptionStyle } from './commonStyles';
+
+type Props = {|
+  +booking: OneWayTrip_booking,
+|};
+
+const OneWayTrip = ({ booking }: Props) => {
+  const origin = idx(booking.trip.departure, _ => _.airport.city.name) || '';
+  const destination = booking.trip.arrival.airport.city.name;
```

Comment:
the last idx 😄

Reply:
@jalbertsr How is it possible that Flow is not able to catch this? Something is really wrong with your codebase...

Reply:
This is one example from the *generated* code:

```
export type OneWayTrip_booking = {|
  +trip: ?{|
    +departure: ?{|
      +airport: ?{|
        +city: ?{|
          +name: ?string,
        |},
      |},
    |},
    +arrival: ?{|
      +airport: ?{|
        +city: ?{|
          +name: ?string,
        |},
      |},
    |},
  |},
  +$refType: OneWayTrip_booking$ref,
|};

```